### PR TITLE
docs: clarify compression performance tradeoffs

### LINKED ISSUE #330: 3x speedup compression claim is false/outdated
Final: the compression claim is inverted on your GB10

Qwen2.5-32B-Instruct, 10 tokens, same prompt, run sequentially:

┌──────┬─────────┬─────────────┬──────────┬─────────────┬───────┐
│ mode │ s/token │ vs baseline │ peak GPU │ bytes/layer │ split │
├──────┼─────────┼─────────────┼──────────┼─────────────┼───────┤
│ none │ 13.3    │ —           │ 1.60GB   │ 975MB       │ 62G   │
├──────┼─────────┼─────────────┼──────────┼─────────────┼───────┤
│ 8bit │ 50.5    │ 3.8x slower │ 3.16GB   │ 489MB       │ 31G   │
├──────┼─────────┼─────────────┼──────────┼─────────────┼───────┤
│ 4bit │ 106.7   │ 8.0x slower │ 2.04GB   │ 274MB       │ 18G   │
└──────┴─────────┴─────────────┴──────────┴─────────────┴───────┘

All three produced correct output — speed and memory only, no accuracy issue.

Against the repo: README:142 claims "up to 3x" speedup; utils.py:498 claims "4x or 2x". Measured: 8x and 3.8x slower, and every compressed mode used more memory than baseline. Plus a one-time 361s split build for 4bit.

Why, and the 4bit run is what proves it: 4bit reads the least data — 274MB/layer, 3.6x less than baseline, entire 18GB split fitting in your 76GB page cache. If the win were I/O, it should have been fastest. It was slowest. Speed tracks dequantization work, not bytes read. uncompress_layer_state_dict (utils.py:86-114) expands every layer back to fp16 per token — files shrink, compute doesn't. 4bit's NF4 path additionally rebuilds a QuantState per tensor through the O(n²) key scan at line 92, costing more than 8bit's simpler path despite reading less.

I'd call this a stale 2023 benchmark, not a dishonest one. On airllm 2.0-era disks, paying dequant cost to read a third of the bytes was plausibly a real win. On NVMe witre nearly free and that overhead is pure loss.

These numbers cleanly contradict a headline README claim, with the causal code path identified. Happy to write it up.

### PR DESCRIPTION
**Pull request by aryansk**
Fixes #330. Summary: remove stale guaranteed speedup numbers from the README and API documentation, explain that compression can reduce storage while adding decompression work, and direct users to benchmark on their own hardware. Validation: python3 -m py_compile air_llm/airllm/utils.py; git diff --check. The change is documentation-only and does not alter runtime behavior.

### CODE REVIEW

**Comment by Hotragn** [comment]
Good direction — dropping an unconditional speedup number is right, and "benchmark on your own hardware" is honest.

One thing that might make the docs more useful than "it depends": the direction isn't hardware-random, it's predictable from what compression actually does. I dug into this on #330 and there are two concrete costs, both in source:

1. **Compression force-disables prefetching** — [`airllm_base.py:172-174`](https://github.com/lyogavin/airllm/blob/main/air_llm/airllm/airllm_base.py#L172-L174):
   ```python
   if self.compression is not None and self.prefetching:
       print("prefetching is not supported together with compression for now; disabling prefetching.")
       self.prefetching = False
   ```
   So you lose the disk/compute overlap entirely.

2. **Every layer is dequantized on the GPU, per layer per token** — `load_layer` (`utils.py:142`) → `uncompress_layer_state_dict` (`utils.py:86-114`): a `.cuda()` upload of the quantized tensor plus a dequant kernel, ~64 layers × every token. 4-bit's `dequantize_nf4` + `QuantState.from_dict` is heavier than 8-bit's `dequantize_blockwise`, which is why TyrelCB measured 8bit landing *between* none and 4bit rather than the two compressed modes being similar.

So the rule users can actually apply is: **compression trades disk bytes for GPU work — it wins when disk read is your bottleneck, and loses when it isn't.** That's why the original "3x" held on 2023-era commodity disks and inverts on a GB10 with fast unified memory. Same for the peak-VRAM surprise in that table (8bit 3.16GB vs none 1.60GB): dequantize materialises the fp16 tensor alongside the quantized source and its quant state.

Might be worth a sentence to that effect so people can predict which side they're on before spending a 62GB split. Entirely your call on wording — just offering the mechanism since I'd already traced it.

(Caveat: I verified the mechanism by reading the code, not by reproducing the benchmark — no GPU here. The magnitudes are TyrelCB's.)

**Comment by aryansk** [comment]
Thanks for tracing the tradeoff through the source and for the concrete explanation. I added a sentence to the README in `4883a6f` covering the current prefetching limitation and per-layer GPU dequantization, while keeping the performance claim qualified and not repeating benchmark magnitudes I could not reproduce without a GPU. `git diff --check` passes; I appreciate the pointer.

**Comment by aryansk** [comment]
Thanks for tracing the mechanism so clearly. I’ve incorporated that explanation into the branch: the README now calls out prefetching being disabled, per-layer GPU dequantization, the disk-I/O tradeoff, and benchmarking on target hardware; the API documentation was updated too. Runtime behavior is unchanged.
